dashbuilder/dashbuilder-shared/dashbuilder-js/packages/time-series-component/src/Chart.tsx

```diff
@@ -0,0 +1,222 @@
+  LOADING_COMPONENT = "Loading Component",
+  LOADED_COMPONENT = "Loaded Component",
+  FINISHED = "Finished loading"
+}
+
+interface AppState {
+  state: AppStateType;
+  processesoptions: Options;
+  processesseries: SingleSeries[];
+  configurationIssue: string;
+  message?: string;
+}
+
+export const validateParams = (params: Map<string, string | number>): string | undefined => {
+  if (!params.get(Params.TRANSPOSED)) {
+    return "Transposed is required.";
+  }
+  if (!params.get(Params.CHARTNAME)) {
+    return "Chart name is required.";
+  }
+};
+
+const validateDataSet = (ds: DataSet, transposed: boolean): string | undefined => {
+  return transposed ? validateTransposedDataset(ds) : validateNonTransposedDataset(ds);
+};
+
+export const validateNonTransposedDataset = (ds: DataSet): string | undefined => {
+  if (ds.columns.length < 2) {
+    return NOT_ENOUGH_COLUMNS_MSG;
+  }
+  for (let i = 1; i < ds.columns.length; i++) {
+    if (ds.columns[i].type !== ColumnType.NUMBER) {
+      return "Wrong type for column " + (i + 1) + ", it should be NUMBER";
+    }
+  }
+};
+
+export const validateTransposedDataset = (ds: DataSet): string | undefined => {
+  if (ds.columns.length < 3) {
+    return NOT_ENOUGH_COLUMNS_MSG_1;
+  }
+  if (ds.columns[1].type !== ColumnType.TEXT && ds.columns[1].type !== ColumnType.LABEL) {
+    return SECOND_COLUMN_INVALID_MSG_1;
+  }
+  if (ds.columns[2].type !== ColumnType.NUMBER) {
+    return THIRD_COLUMN_INVALID_MSG_1;
+  }
+};
+
+interface Props {
+  controller: ComponentController;
+}
+
+function getSeries(dataset: DataSet, transposed: boolean): SingleSeries[] {
+  return transposed ? getSeriesforTransposedDataset(dataset) : getSeriesforNonTransposedDataset(dataset);
+}
+
+export function getSeriesforNonTransposedDataset(dataset: DataSet): SingleSeries[] {
+  const arrayseries: SingleSeries[] = [];
+  let newseries: SingleSeries = { name: "", data: [] };
+  for (let i = 1; i < dataset.columns.length; i++) {
+    newseries = {
+      name: dataset.columns[i].name,
+      data: dataset.data.map((d: Array<string | number>) => +d[i])
+    };
+    arrayseries.push(newseries);
+  }
+  return arrayseries;
+}
+
+export function getSeriesforTransposedDataset(dataset: DataSet): SingleSeries[] {
+  const arrayseries: SingleSeries[] = [];
+  let newseries: SingleSeries = { name: "", data: [] };
+  let names: string[] = [];
+  let datum: Array<string | number> = [];
+  const newnames: string[] = [];
+  names = dataset.data.map((d: string[]) => d[1]);
+  datum = dataset.data.map((d: Array<string | number>) => d[2]);
+  for (const i of names) {
+    if (newnames.indexOf(i) === -1) {
+      newnames.push(i);
+    }
+  }
+  for (const i of newnames) {
+    const newdata: Array<number | string> = [];
+    for (let j = 0; j < names.length; j++) {
+      if (i === names[j]) {
+        newdata.push(+datum[j]);
+      }
+    }
+    newseries = { name: i, data: newdata };
+    arrayseries.push(newseries);
```

Comment:
```suggestion
    arrayseries.push({ name: i, data: newdata });
```
With this, you wouldn't need the variable `newseries`.